src/comms.rs

```diff
@@ -163,6 +167,7 @@ impl Gate {
     /// This method will send out the update to all active links. It will
     /// also update the gate metrics based on the update.
     pub async fn update_data(&mut self, update: payload::Update) {
+        self.unit_data = Some(update.clone());
```

Comment:
Woah! Slow down! ;)

Startup issues is one reason why the gate keeps the last update, but the other one is a (currently theoretical) dynamic reconfiguration. If you hook up a new link later, it will also get the last update when it is subscribing.

Just keeping it until you have the first subscriber isn’t good enough – there may be more subscribers coming after the first one and they would need the update, too.

`unit_data` will indeed only be `None` before the first update has been submitted to the gate. I considered making the initialization phase more explicit, but that results in messy unit states (`Gone` is already pretty terrrible). So now `UnitState::Healthy` and no update means that the unit is currently bootstrapping. I might rethink all these unit states in a separate PR but for now I didn’t want to change too much all at once.

The new `get_data` method allowing access to the current update is actually on purpose. The _any_ unit currently keeps copies of all the updates so it can seamlessly switch. With this change, it doesn’t need to any more but can just get the current copy from the link if it needs it. The upcoming _merge_ unit will also have to keep all updates from all sources available.

I did consider changing the return type of `Link::query` to only indicate that there is a new update and have you then go and ask if you want it. But since it now returns a reference to the stored update, this is actually a shortcut just for that which is quite convenient since a lot of components will actually want to do something with the update anyway.

Reply:
Regarding future dynamic reconfiguration, if the last update isn't time sensitive, i.e. doesn't need to be processed around the time it is generated, then this will work. I'm less sure about the defined interface with these changes to make updates available via two different mechanisms but let's leave it like this for now as it addresses actual issues.